### ISSUE TITLE
CameraHead quaternion output not sign-standardized like GT encoding

### ISSUE THREAD

**Issue by AsherJingkongChen** [facebookresearch/vggt-omega#25]
`extri_intri_to_pose_encoding` standardizes the GT quaternion to `w >= 0`, but CameraHead outputs a raw quaternion (`w < 0` in the released 1B weights).

The L1 loss of pose encoding then sees `~1.5` error on `encoding[..., 6:7]` for identical rotations (q vs -q, double cover). Should CameraHead standardize its quaternion output to match the GT convention?

**Comment by jytime** [comment]
Hi @AsherJingkongChen ,

I tried both and it seems did not affect a lot from my observation. If you are worried about it, feel free to use the code below:

```

    pred_q = pred_enc[..., component_slice]
    gt_q = gt_enc[..., component_slice]
    
    # Compute L1 distance to both +q and -q
    # Sum over the quaternion components (dim=-1) to get total distance per sample
    dist_plus = (pred_q - gt_q).abs().sum(dim=-1) # BxS
    dist_minus = (pred_q + gt_q).abs().sum(dim=-1) # BxS
    
    # Choose the sign that results in lower L1 distance
    # shape: (..., 1) after unsqueeze
    mask = (dist_plus < dist_minus)[..., None] # BxSx1
    
    # If dist_plus < dist_minus, use +gt, else use -gt
    gt_target_q = torch.where(mask, gt_q, -gt_q) # BxSx4
    
    loss_q = (pred_q - gt_target_q).abs() # BxSx4
    
    
```

**Comment by AsherJingkongChen** [comment]
@jytime Thanks, that clears it up. On the pretrained weights we use downstream, I went with a dot-product sign flip instead of standardizing. Align the GT quaternion to the prediction with $sign(<\hat{g}, g>)$, then plain L1. It stays smooth at the 180 degree boundary. On the other hand, `standardize-to-(w>=0)` jumps there because the flip triggers on the prediction's own `w`.

Sharing my snippet in case it helps:

```python
def align_pose_encoding(target, reference):
    quat = target[..., 3:7]
    quat = quat.where(quat.mul(reference[..., 3:7]).sum(dim=-1, keepdim=True) >= 0.0, -quat)
    return torch.cat([target[..., :3], quat, target[..., 7:]], dim=-1)
```

Thanks again!
